src/utils/profileUtils.ts

```diff
@@ -115,6 +117,22 @@ export function isEthAddress(address: string | null) {
   return address?.match(/^(0x)?[0-9a-fA-F]{40}$/);
 }
 
+export async function lookupAddressWithRetry(
+  web3Provider: Provider,
+  address: EthereumAddress
+) {
+  let count = 0;
+  while (count < 3) {
+    try {
+      const ens = await web3Provider.lookupAddress(address);
+      return ens;
+    } catch (e) {
+      count += 1;
+    }
+  }
+  return null;
```

Comment:
Maybe we can make this part a bit cleaner? There are a couple things to consider:
- `(e)` part in `catch` is optional (or at least usually, maybe in our case it breaks but idk).
- how about `for` loop instead of `while`?
- we can avoid variable assignment and `return` directly (should work the same, but again needs to be checked).

```js
for (let i = 0; i < 3; i++) {
  try {
    return await web3Provider.lookupAddress(address);
  } catch {}
}
```

Reply:
this def cleaner but otherwise 👌